sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -256,7 +256,7 @@ func (c *control) handleInstruction(ctx context.Context, req *fnpb.InstructionRe
 		}
 
 		// Get the desired splits for the root FnAPI read operation.
-		ds := msg.GetDesiredSplits()["0"]
+		ds := msg.GetDesiredSplits()[plan.Source().SID.Target.ID]
```

Comment:
This will collide with Luke's https://github.com/apache/beam/pull/8733 , as a headsup.
cc: @lukecwik